lib/services/protocol_service.ts

```diff
@@ -572,11 +572,30 @@ export class SNProtocolService extends PureService implements EncryptionDelegate
         continue;
       }
       try {
+        let keyForPayload = key;
+        const version = encryptedPayload!.version;
+        const payloadContentType = encryptedPayload!.content_type;
+        const keyContentType = key?.content_type;
+        /** We check if the key can decrypt the encrypted payload. */
+        if (
+          version === ProtocolVersion.V004 &&
```

Comment:
This needs to be more future proof. Imagine that V005 comes out one month later with a minor change. Do we keep adding if conditions here for every new version?